src/env/from_envp.c

```diff
@@ -13,13 +13,12 @@
 #include <assert.h>
 #include "private.h"
 
-static void		free_new(struct s_env_pair	*new)
+static int		free_new(struct s_env_pair	*new)
 {
-	ft_memdel((void **)new);
-	if (new->key != NULL)
-		ft_strdel(&new->key);
-	if (new->value != NULL)
-		ft_strdel(&new->value);
+	ft_strdel(&new->key);
+	ft_strdel(&new->value);
+	ft_memdel((void **)&new);
+	return (-1);
```

Comment:
this faux return (ie always returning -1) due to a higher-level function having line-trouble is not acceptable.